lib/gauge_fix_ovr.cu

```diff
@@ -1530,62 +1367,26 @@ public:
-	  #else
-	  printfQuda("Time: %6.6f s, Gflop/s = %6.1f, GB/s = %6.1f\n", secs, gflops, gbytes);
-	  #endif
-	}
-  }
-
-  template<typename Float, int NElems, typename Gauge>
-  void gaugefixingOVR( Gauge dataOr,  cudaGaugeField& data, const int gauge_dir, const int Nsteps, const int verbose_interval,
-                       const Float relax_boost, const double tolerance, const int reunit_interval, const int stopWtheta) {
-    if ( gauge_dir != 3 ) {
-      printfQuda("Starting Landau gauge fixing...\n");
-      gaugefixingOVR<Float, Gauge, NElems, 4>(dataOr, data, Nsteps, verbose_interval, relax_boost, tolerance, reunit_interval, stopWtheta);
-    }
-    else {
-      printfQuda("Starting Coulomb gauge fixing...\n");
-      gaugefixingOVR<Float, Gauge, NElems, 3>(dataOr, data, Nsteps, verbose_interval, relax_boost, tolerance, reunit_interval, stopWtheta);
+      double gflops = (flop * 1e-9) / (secs);
+      double gbytes = byte / (secs * 1e9);
+      printfQuda("Time: %6.6f s, Gflop/s = %6.1f, GB/s = %6.1f\n", secs, gflops * comm_size(), gbytes * comm_size());
     }
   }
 
-
-
-  template<typename Float>
-  void gaugefixingOVR( cudaGaugeField& data, const int gauge_dir, const int Nsteps, const int verbose_interval,
-		       const Float relax_boost, const double tolerance, const int reunit_interval, const int stopWtheta) {
-
-    // Switching to FloatNOrder for the gauge field in order to support RECONSTRUCT_12
-    if ( data.isNative() ) {
-      if ( data.Reconstruct() == QUDA_RECONSTRUCT_NO ) {
-        //printfQuda("QUDA_RECONSTRUCT_NO\n");
-        numParams = 18;
-	typedef typename gauge_mapper<Float,QUDA_RECONSTRUCT_NO>::type Gauge;
-        gaugefixingOVR<Float, 18>(Gauge(data), data, gauge_dir, Nsteps, verbose_interval, relax_boost, tolerance, reunit_interval, stopWtheta);
-      } else if ( data.Reconstruct() == QUDA_RECONSTRUCT_12 ) {
-        //printfQuda("QUDA_RECONSTRUCT_12\n");
-        numParams = 12;
-	typedef typename gauge_mapper<Float,QUDA_RECONSTRUCT_12>::type Gauge;
-        gaugefixingOVR<Float, 12>(Gauge(data), data, gauge_dir, Nsteps, verbose_interval, relax_boost, tolerance, reunit_interval, stopWtheta);
-      } else if ( data.Reconstruct() == QUDA_RECONSTRUCT_8 ) {
-        //printfQuda("QUDA_RECONSTRUCT_8\n");
-        numParams = 8;
-	typedef typename gauge_mapper<Float,QUDA_RECONSTRUCT_8>::type Gauge;
-        gaugefixingOVR<Float, 8>(Gauge(data), data, gauge_dir, Nsteps, verbose_interval, relax_boost, tolerance, reunit_interval, stopWtheta);
+  template <typename Float, int nColor, QudaReconstructType recon> struct GaugeFixingOVR {
+    GaugeFixingOVR(GaugeField& data, const int gauge_dir, const int Nsteps, const int verbose_interval,
+                   const Float relax_boost, const double tolerance, const int reunit_interval, const int stopWtheta)
+    {
+      using Gauge = typename gauge_mapper<Float, recon>::type;
+      if (gauge_dir != 3) {
```

Comment:
Technically out of the scope of this PR, but I think this should throw an error if `gauge_dir` isn't strictly 3 or 4.

Reply:
https://github.com/lattice/quda/pull/1046/commits/9a37a41912257a50bc038ae16f48544c4f301fb8